src/test/python/complete_winml_tutorial.py

```diff
@@ -0,0 +1,142 @@
+import find_projection
+
+import pyrt
+import asyncio
+
+def timed_op(fun):
+    import time
+
+    def sync_wrapper(*args, **kwds):
+        print("Starting", fun.__name__)
+        start = time.perf_counter()
+
+        ret = fun(*args, **kwds)
+
+        end = time.perf_counter()
+        print(fun.__name__, "took", end - start, "seconds")
+        return ret
+
+    async def async_wrapper(*args, **kwds):
+        print("Starting", fun.__name__)
+        start = time.perf_counter()
+
+        ret = await fun(*args, **kwds)
+
+        end = time.perf_counter()
+        print(fun.__name__, "took", end - start, "seconds")
+        return ret
+
+    return async_wrapper if asyncio.iscoroutinefunction(fun) else sync_wrapper
+
+async def wrap_async_op(op):
+    loop = asyncio.get_event_loop()
+    future = loop.create_future()
+
+    def callback(operation, status):
+        if status == 1:
+            result = operation.GetResults()
+            loop.call_soon_threadsafe(asyncio.Future.set_result, future, result)
+        elif status == 2:
+            loop.call_soon_threadsafe(asyncio.Future.set_exception, future, asyncio.CancelledError())
+        elif status == 3:
+            loop.call_soon_threadsafe(asyncio.Future.set_exception, future, RuntimeError("AsyncOp failed"))
+        else:
+            loop.call_soon_threadsafe(asyncio.Future.set_exception, future, RuntimeError("Unexpected AsyncStatus"))
+
+    op.put_Completed(callback)
+
+    return await future
+
+def run_async_code(code):
+    loop = asyncio.get_event_loop()
+    loop.run_until_complete(code())
+    loop.close()
+
```

Comment:
this all looks like a generally useful module

Reply:
run_async_code is a Python 3.6 version of the [asyncio.run method](https://docs.python.org/3/library/asyncio-task.html#asyncio.run) from Python 3.7. Example Code: https://docs.python.org/3/library/asyncio-task.html#coroutines

timed_op is useful. I wrote the original version *years ago* when I was working on IronPython (though I did have to add async support for this code). Nothing to do w/ WinRT/Xlang tho.

wrap_async_op addressed below